protocol_grpc.go

```diff
@@ -696,6 +701,39 @@ func grpcValidateResponse(
 	return nil
 }
 
+func grpcVerifyTrailersOnly(response *http.Response) *Error {
+	// Make sure there's nothing in the body.
+	if numBytes, err := discard(response.Body); err != nil {
+		err = wrapIfContextError(err)
+		if connErr, ok := asError(err); ok {
+			return connErr
+		}
+		return errorf(CodeInternal, "corrupt response: I/O error after trailers-only response: %w", err)
+	} else if numBytes > 0 {
+		return errorf(CodeInternal, "corrupt response: %d extra bytes after trailers-only response", numBytes)
+	}
```

Comment:
This was largely copied from the handling of end-stream messages (in `envelopeReader.Unmarshal`), where we also want to verify that there is no subsequent response body data.